dsa-pros/leetcode/array/JavaScript/JuwonAdeyemi/ArrayWeek4.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @param {number[]} nums
+ * @return {boolean}
+ *
+ * Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
+ */
+var containsDuplicate = function (nums) {
+  nums.sort();
+  for (let i = 0; i <= nums.length - 1; i++) {
+    if (nums[i] === nums[i + 1]) {
+      return true;
+    }
+  }
+  return false;
+};
+
+/**
+ * @param {number[]} nums
+ * @param {number} target
+ * @return {number}
+ * 
+ * Given an array of integers nums which is sorted in ascending order, and an integer target, write a function to search target in nums. If target exists, then return its index. Otherwise, return -1.
+
+You must write an algorithm with O(log n) runtime complexity.
+
+ 
+ */
+var search = function (nums, target) {
+  nums.sort;
+  let i = nums.indexOf(target);
+  return i;
+};
```

Comment:
Irene stated a condition that you missed here

Reply:
Also the question requires that `You must write an algorithm with O(log n) runtime complexity`. This solution does not satisfy that requirement even though it works. You might want to check that out

Reply:
> Also the question requires that `You must write an algorithm with O(log n) runtime complexity`. This solution does not satisfy that requirement even though it works. You might want to check that out

Okay
Done, Thank you

Reply:
This solution is O(n) time complexity.  Look up Binary search and view some examples with various solutions as Mary suggested, will help a lot in understanding.